param/parameterized.py

```diff
@@ -1720,17 +1718,25 @@ def _events(self_, value):
         self_.self_or_cls._param__private.parameters_state['events'] = value
 
     @property
-    def _watchers(self_):
+    def _state_watchers(self_):
         return self_.self_or_cls._param__private.parameters_state['watchers']
 
-    @_watchers.setter
-    def _watchers(self_, value):
+    @_state_watchers.setter
+    def _state_watchers(self_, value):
         self_.self_or_cls._param__private.parameters_state['watchers'] = value
 
     @property
-    def watchers(self):
-        """Read-only list of watchers on this Parameterized"""
-        return self._watchers
+    def watchers(self_):
+        """Dictionary of instance watchers."""
+        if self_.self is None:
+            raise TypeError()
+        return self_.self._param__private.watchers
```

Comment:
Briefly considered whether this should be a `FrozenDict` copy of the real dict but I think being able to modify the watchers is kind of the point.

Reply:
Yep indeed I thought you told me that you were manipulating the watchers in Panel. Also if I'm right there's no frozendict implementation in the standard library.

Good anyway that you commented right there, the `TypeError` raised in line 1732 is a little dry :) I'm going to merge after fixing that.